code/src/cljs/sixsq/nuvla/ui/notifications/events.cljs

```diff
@@ -0,0 +1,458 @@
+(ns sixsq.nuvla.ui.notifications.events
+  (:require
+    [cljs.spec.alpha :as s]
+    [re-frame.core :refer [dispatch dispatch-sync reg-event-db reg-event-fx]]
+    [sixsq.nuvla.ui.cimi-api.effects :as cimi-api-fx]
+    [sixsq.nuvla.ui.cimi-detail.events :as cimi-detail-events]
+    [sixsq.nuvla.ui.notifications.spec :as spec]
+    [sixsq.nuvla.ui.notifications.utils :as utils]
+    [sixsq.nuvla.ui.messages.events :as messages-events]
+    [sixsq.nuvla.ui.utils.response :as response]))
+
+;;
+;; notification-methods
+;;
+
+(reg-event-db
+  ::method
+  (fn [db [_ method]]
+    (assoc-in db [::spec/notification-method :method] method)))
+
+
+(reg-event-db
+  ::set-notification-method
+  (fn [db [_ notification-method]]
+    (assoc db ::spec/notification-method notification-method)))
+
+
+(reg-event-db
+  ::set-notification-methods
+  (fn [db [_ notification-methods]]
+    (assoc db ::spec/notification-methods notification-methods)))
+
+
+(reg-event-fx
+  ::get-notification-methods
+  (fn [{:keys [db]} [_]]
```

Comment:
[_] should be replaced by _
Like that clojurescript will do the destructuring with first

Reply:
changed